Fcom 226

KRuusuv opened the pull request:
A small collection of tasks for consuming Nordea's web service. The tasks contained are: GetUserInfo, DownLoadFileList, UploadFile and DownloadFile. GetUserInfo has not yet been tested, but the other tasks have been and have worked as intended.

Commit:
FCOM-226, first commit, task version 1.0.8

Commit:
FCOM-226, updated readme and some comments in task code. Removed bin files.

Commit:
FCOM-226, updated README.md

OssiGalkin commented:
## Documentation:

Format of dates is not defined. (it can be seen from the example, but its newer hurt to write it down)

päivämäärät voisi kertoa muotonsa (se kullä näkyy esimerkissä mutta silti)

This could be simplified:
"Valid values for files sent to the bank by the customer are "WFP" or "FWD" (WFP = waiting for processing. FWD = forwarded). Valid values for files sent to the bank by the customer are "NEW" or "DLD" (NEW = files not downloaded yet. DLD = files already downloaded). "

Is GetUserInfo tested yet? If not it could be moved to the end of the documentation.

It could be said what cert is used.

## Code:

Helper.cs line 40:  securityNode can be null (SelectSingleNode on the previous line might return it), thus this line might throw an error.

GetSignatureNode on line 233, could take argument referenceIds as IList<string> (currently  IEnumerable<string> ). It is not generally safe to enumerate IEnumerable twice.

MessageService.cs contains OP bank only things.

#pragma warning disable 1591 could be applied more often, now building throws dozens of XML warnings

Commit:
FCOM-226, a number of small fixes to documentation and code

Commit:
FCOM-226, A number of small fixes to documentation and code

KRuusuv commented:
Documentation:

-Format of dates is now defined.
-Fixed explanation of valid status values.
-GetUserInfo has been tested. Works, but response could be made neater in a later iteration.
-Some certification type descriptions added, as well as a mention of Nordea root certificate.

Code:

-Added null check for securityNode. En exception will be thrown if this is left null with a simple error message.
-IEnumerable changed to IList for GetSignatureNode.
-Removed OP bank related things from MessageService.cs.
-Added a bunch of #pragma warning disable 1591s.

OssiGalkin committed:
Updated documentation, increased version number, and added .gitignore.
